### PR TITLE
Publish llms.txt and llms-full.txt for skillsaw.org

### LINKED ISSUE #416: Publish llms.txt and llms-full.txt for skillsaw.org
skillsaw.org is built by `scripts/generate-site-content.py` (674 lines), which already machine-generates 68 rule/group pages plus `docs/cli.md` and `docs/research.md` before every `mkdocs build`. That means an `llms.txt` index and a concatenated `llms-full.txt` are nearly free: a couple of new generator functions reusing the existing per-page collectors, and mkdocs copies non-`.md` files in `docs/` verbatim to the site root (the `docs/CNAME` file already relies on this), so `docs/llms.txt` becomes `skillsaw.org/llms.txt` with zero build wiring.

This is unusually on-brand for skillsaw: the primary audience of the docs is literally coding agents. `docs/getting-started.md` (line ~60) tells agents to fetch a raw `SKILL.md` URL from GitHub, and `skills/skillsaw-create-plugin/SKILL.md` (lines 23–25) sends agents to skillsaw.org pages that today serve only HTML. One fetch of `llms-full.txt` should let any agent answer "how do I configure content-weak-language" without scraping HTML.

## Why

- Zero `llms.txt` / `llms-full.txt` support exists today: no hits in `mkdocs.yml`, `docs/`, or `scripts/`, and no `mkdocs-llmstxt` plugin in `pyproject.toml`. Every `llms` grep hit is prose about LLMs (`docs/research.md`, rule pages).
- The content is already generated and committed as markdown: `docs/rules/` holds 68 rule/group pages plus `index.md`, all produced by `generate-site-content.py` with reusable per-page functions (`generate_rule_page`, `generate_group_page`, `generate_cli_reference`, `generate_research_page`).
- The deploy pipeline needs no changes: `.github/workflows/pages.yml` runs `python scripts/generate-site-content.py` and then `mkdocs build`, so anything the generator drops into `docs/` ships automatically.
- skillsaw's own skills and docs direct agents at these URLs, so the site's markdown-consumability is part of the product, not a nice-to-have.

## Proposed implementation

1. Add `generate_llms_txt()` and `generate_llms_full()` to `scripts/generate-site-content.py` (~80–100 lines total), called from `main()` alongside the existing generators.
2. `llms.txt` (the index): a short project blurb plus sectioned links to guides and rule pages. Link to `raw.githubusercontent.com` URLs of the committed `.md` sources — mkdocs does not serve per-page markdown, so linking site HTML pages would defeat the purpose. All `docs/*.md` and `docs/rules/*.md` sources are committed, so raw URLs are stable.
3. `llms-full.txt`: concatenate the guides plus all rule pages, excluding `docs/designs/` (internal design docs). Include a small markup-cleanup pass that strips mkdocs-material tab markers (`=== "..."` — there are exactly 7 in `docs/getting-started.md`) and admonition syntax, so the output is plain markdown any agent can consume.
4. Decide committed vs. gitignored for the ~246 KB of generated output. Existing generated docs pages are committed (`make update` runs the generator), so committing would match convention — but it adds a large diff on every rule change. Recommendation: add both files to `.gitignore` and let `pages.yml` produce them at deploy time; it already runs the generator before `mkdocs build`, so nothing else changes.
5. Mention the new endpoints in `README.md` and `docs/index.md` per the pre-PR checklist.

Gotchas surfaced during verification:

- Don't reach for the `mkdocs-llmstxt` plugin — the in-house generator already owns all page content, and a plugin adds a dependency for something ~100 lines cover.
- The cleanup pass in step 3 is what keeps `llms-full.txt` useful; raw concatenation would leak mkdocs-material syntax that reads as noise to a model.
- No test infrastructure exists for `generate-site-content.py` (nothing in `tests/` references it), so a heavyweight test suite isn't precedented here; a smoke check that the two files are generated and non-empty is proportionate.

Files to touch:

- `scripts/generate-site-content.py`
- `docs/llms.txt` (generated)
- `docs/llms-full.txt` (generated)
- `.gitignore`
- `README.md`
- `docs/index.md`
- `Makefile` — likely no change; the `generate-site-content` target already covers it

## Scope

**Effort: S** — roughly a day: two generator functions, a markup-cleanup pass, the gitignore decision, and doc mentions.

Out of scope:

- Serving per-page markdown variants of every docs page (e.g. `.md` alongside each HTML page).
- Adopting the `mkdocs-llmstxt` plugin or any new dependency.
- Any MCP-based docs access (previously explored in the now-closed #403).

Interactions with open PRs: #410 also touches `scripts/generate-site-content.py`, but only to regenerate docs for a new rule — the additions here are new functions and merge trivially. #408 and #409 modify guide content (`docs/getting-started.md`, `docs/cli.md`, `docs/autofixing.md`); `llms-full.txt` is generated from whatever is on disk, so it picks those changes up automatically with no conflicts. #407 is already merged and touches only build tooling.

---

Surfaced and adversarially verified by a multi-agent codebase exploration. 🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by rajusem**
Fixes #416.

skillsaw.org's primary audience is coding agents — the repo's own skills send agents to skillsaw.org pages that today serve only HTML. This adds the two conventional plain-text endpoints, generated by `scripts/generate-site-content.py` at deploy time with zero build wiring: mkdocs already copies non-`.md` files in `docs/` verbatim (the `docs/CNAME` precedent), and `pages.yml` already runs the generator before `mkdocs build`.

## What it does

- **`skillsaw.org/llms.txt`** — a curated llms.txt-format index: project summary, the guides in nav order, the 16 rule groups with one-line blurbs, and every canonical rule page under `## Optional`, linked as `raw.githubusercontent.com` sources (the site serves no per-page markdown; every linked source is committed). Deprecated rules are labeled with their deprecation version.
- **`skillsaw.org/llms-full.txt`** — the full docs as one plain-markdown file (~388 KB / ~93k tokens): one fetch answers "how do I configure content-weak-language". The cleanup pass flattens mkdocs-material tabs and admonitions to bold labels and dedents their bodies, strips icon shortcodes and attr lists (numeric step icons become "N." so the onboarding table keeps its sequence), rewrites in-page `---` horizontal rules to `***` so `---` unambiguously means "next page", and preserves code fences verbatim — including fences indented inside tab bodies and the 4-backtick-wrapping-unclosed-fence example.
- The marketing homepage is excluded from the full file (it is raw HTML/frontmatter) and honestly labeled "(HTML-heavy hero page)" in the index; the `context-budget` group/rule slug collision — a pre-existing generator quirk where the rule page overwrites the group page at the same path — is deduplicated in both outputs, not fixed here.
- Both files are **gitignored and regenerated at deploy**, per the issue's own recommendation: ~400 KB of churn stays out of every rule-edit diff and `verify-update` stays green.
- A fail-loud `guides_coverage_errors()` check (mirroring the existing `RULE_GROUPS` guard) fails generation whenever `GUIDES` and the on-disk `docs/*.md` drift, so the index can never silently miss a new guide.

## Tests

The issue suggests a smoke check; the suite is somewhat larger for one reason — because the outputs are gitignored, there is no committed diff to catch a flattening regression, so the transform's edge cases are pinned from the real corpus (fence-in-tab dedent, back-to-back tabs, bare admonitions with immediate bodies, deprecated-page shape, setext-underline vs horizontal rule, idempotency), plus llms.txt structure, once-per-rule coverage, the collision dedup, missing-page fail-loud, and the `main()` wiring.

- `make test`: 4,533 passed
- `make lint` clean; `make update` clean (self-lint A+, 0 errors, 0 warnings); regeneration is byte-identical run-to-run
- `mkdocs build` ships both files to `site/`; the raw URL form was verified live (HTTP 200, `text/plain`)

Out of scope, per the issue: no `mkdocs-llmstxt` plugin or new dependencies, no per-page markdown variants, no MCP docs access.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added AI-friendly documentation indexes and complete plain-Markdown documentation.
  * Added homepage and README links for discovering documentation intended for AI agents.
  * Documentation outputs are automatically generated from the site content and navigation.

* **Tests**
  * Added coverage for documentation generation, formatting conversion, validation, error handling, and repeatable output generation.

* **Chores**
  * Excluded generated AI documentation files from version control.

### CODE REVIEW

File: scripts/generate-site-content.py [hunk 1]
```diff
@@ -747,6 +753,276 @@ def inject_stats(index_path, rules_data):
...
+            out.append(line)
+            continue
+        if line.strip() == "---" and (not out or not out[-1].strip()):
+            # In-page horizontal rule (preceded by a blank line, so not a
+            # setext heading): rewrite to the equivalent `***` so `---`
+            # stays reserved for the page joins in llms-full.txt.
+            out.append("***")
+            continue
+        out.append(_strip_inline_noise(line))
+    return "\n".join(out)
```
[the remark this answers is not in the document]
Comment (rajusem) [reply scripts/generate-site-content.py:942]: The fence defect is real — reproduced before fixing: a non-bare ```bash line inside an open fence closed it, so later markers inside the block could be flattened as prose. Applied the minimal spec-correct fix in 461094b: a closing fence must be bare (verified against markdown-it-py in commonmark mode — an info string disqualifies a close for backtick and tilde fences alike), plus a regression test using this exact scenario. The generated outputs are byte-identical before and after, confirming the defect was latent.

Declining the MarkdownDoc rewrite itself: MarkdownDoc parses strict CommonMark, which cannot represent the pymdownx syntax this transform exists to flatten — markdown-it-py classifies 4-space tab bodies as indented code and `=== "..."` markers as paragraph text — so the block state machine would survive any rewrite; the AST would sit underneath it without removing anything. The cited guideline is also framed for lint rules reading lint-tree blocks (`block.markdown` is a lint-tree accessor); this is a standalone docs-generation script outside that pipeline.

File: scripts/generate-site-content.py [hunk 2]
```diff
@@ -747,6 +753,276 @@ def inject_stats(index_path, rules_data):
...
+
+_NUMERIC_ICON_RE = re.compile(r":material-numeric-(\d+)-circle:")
+
+
+def _strip_inline_noise(line):
+    """Remove mkdocs-material attr lists and icon shortcodes from a prose line."""
+    original = line
+    line = _NUMERIC_ICON_RE.sub(r"\1.", line)
+    line = _ATTR_LIST_RE.sub("", line)
+    line = _ICON_RE.sub("", line)
```
[the remark this answers is not in the document]
Comment (rajusem) [reply scripts/generate-site-content.py]: Applied in 461094b: stripping now runs per prose segment and inline code spans are preserved verbatim, with a regression test pinning a literal `:sparkles:` and `{ .step-icon }` in backticks surviving while the same tokens in surrounding prose are stripped. Hardened further in 127bed1 with a CommonMark code-span scanner (a span closes at a run of exactly the opener's length), so multi-backtick spans quoting backtick-containing content are covered too. Outputs are byte-identical throughout — no such span exists in the docs today.

File: scripts/generate-site-content.py [hunk 3]
```diff
@@ -747,6 +753,276 @@ def inject_stats(index_path, rules_data):
...
+    )
+    parts = [header]
+    for relpath in paths:
+        page = docs_dir / relpath
+        if not page.exists():
+            raise FileNotFoundError(
+                f"docs/{relpath} is missing — run the page generators before "
+                "generate_llms_full()"
+            )
+        parts.append(_plain_markdown(page.read_text(encoding="utf-8")).strip())
```
[the remark this answers is not in the document]
Comment (rajusem) [reply scripts/generate-site-content.py:1057]: Declining this one for this PR: the relative links are a consciously accepted trade-off from the design review. The llms.txt index links are all absolute raw-source URLs already; inside llms-full.txt every relative target's content is present in the same file under a matching H1, so a reader resolves by name without fetching — and llmstxt.org does not require link rewriting in the full file. A correct rewriter is genuinely follow-up-sized: it needs per-page base-dir resolution (docs/ vs docs/rules/), anchor and absolute-link skipping, and code-span awareness to avoid rewriting link syntax inside examples. Happy to file it as a follow-up issue so it gets a tracked decision.

File: scripts/generate-site-content.py [hunk 4]
```diff
@@ -747,6 +753,299 @@ def inject_stats(index_path, rules_data):
...
+_TAB_MARKER_RE = re.compile(r'^=== "(.+)"\s*$')
+_ADMONITION_RE = re.compile(r'^(?:!!!|\?\?\?)\+? +([\w-]+)(?: +"(.*)")?\s*$')
+_ATTR_LIST_RE = re.compile(r"\{:? *\.[\w][\w .-]*\}")
+_ICON_RE = re.compile(r":([a-z][a-z0-9_+-]*):")
+
+
+_NUMERIC_ICON_RE = re.compile(r":material-numeric-(\d+)-circle:")
+
+
+_CODE_SPAN_RE = re.compile(r"`+[^`]+`+")
```
[the remark this answers is not in the document]
Comment (rajusem) [reply scripts/generate-site-content.py]: Applied in 127bed1: replaced the greedy-delimiter regex with a small scanner implementing CommonMark's code-span rule — a span opens with a run of N backticks and closes at the next run of exactly N — so ``literal ` and :sparkles:`` is one span and nothing inside it is stripped; an unmatched run stays plain text. Regression test uses your exact repro line. Outputs remain byte-identical (the shape is absent from the docs today), so this is latent hardening.

File: scripts/generate-site-content.py [hunk 5]
```diff
@@ -747,6 +753,322 @@ def inject_stats(index_path, rules_data):
...
+_ICON_RE = re.compile(r":([a-z][a-z0-9_+-]*):")
+
+
+_NUMERIC_ICON_RE = re.compile(r":material-numeric-(\d+)-circle:")
+
+
+_BACKTICK_RUN_RE = re.compile(r"`+")
+
+
+def _split_code_spans(line):
```
[the remark this answers is not in the document]
Comment (rajusem) [reply scripts/generate-site-content.py:831]: Declining this one: same disposition as the MarkdownDoc suggestion on the sibling thread (which CodeRabbit reviewed and confirmed closed after 461094b), and the argument got stronger, not weaker. MarkdownDoc parses strict CommonMark and cannot represent the pymdownx tab/admonition syntax this transform exists to flatten — the line state machine survives any AST underneath it. On the specific risk: I scanned every page in the concatenation corpus for spans opened on one line and closed on a later line containing icon or attr-list tokens — zero exist, and the generated outputs have stayed byte-identical through every round of this hardening. The worst theoretical outcome of a future multiline span is a cosmetic icon-token removal in prose of our own deploy-time docs, guarded by byte-identical-regeneration checks — not content corruption. Trading that for cross-line span state in a deliberately line-based transform is complexity in the wrong place.

**Comment by stbenjam** [comment]
Thanks!
